src/logic/embedding.lean

```diff
@@ -28,22 +28,32 @@ initialize_simps_projections embedding (to_fun → apply)
 
 end function
 
+section equiv
+
+variables {α : Sort u} {β : Sort v} (f : α ≃ β)
+
 /-- Convert an `α ≃ β` to `α ↪ β`. -/
-@[simps]
-protected def equiv.to_embedding {α : Sort u} {β : Sort v} (f : α ≃ β) : α ↪ β :=
-⟨f, f.injective⟩
+@[simps] protected def equiv.to_embedding : α ↪ β := ⟨f, f.injective⟩
+
+instance equiv.coe_embedding : has_coe (α ≃ β) (α ↪ β) := ⟨equiv.to_embedding⟩
+
+instance equiv.perm.coe_embedding : has_coe (equiv.perm α) (α ↪ α) := equiv.coe_embedding
```

Comment:
This should probably be marked `reducible` since `equiv.perm` is also reducible, otherwise I suspect a tactic might unfold the type but not the `coe` and get itself stuck.
```suggestion
@[reducible]
instance equiv.perm.coe_embedding : has_coe (equiv.perm α) (α ↪ α) := equiv.coe_embedding
```